### DISCUSSION TITLE
issue: Fails to revalidate form on input change via <Controller /> component

### DISCUSSION THREAD

**Discussion by shafkathullah** [react-hook-form/react-hook-form#11706] [Q&A]
### Version Number

^7.31.3

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-revalidate-issue-w4n8gy?file=%2Fsrc%2FApp.tsx%3A79%2C23

### Steps to reproduce

1. Go to CSB link provided
2. Select `Other` from select menu
3. Submit the form, see error message showing correctly
4. Now select any other value from select menu(except 'Select a reason' ofcourse)
5. Error message stays until form is submitted again

### Expected behaviour

1. On 5th point in the steps to reproduce I expect error message to disappear when select input value is changed, as default value for is `reValidateMode` is `onChange`.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shafkathullah** [comment]
Can anyone tell me why this landed in discussion? Isn't it supposed to be in issue?

**Comment by danielnakamashi** [comment]
Hey, your example is working for me. Have you changed anything?

**Comment by shafkathullah** [comment]
@danielnakamashi  No i didn't. specified issue is still there. inshort value change only trigger revalidation for that particular field, not every relevent field. we will have to manually trigger those validation.

**Comment by shafkathullah** [comment]
@danielnakamashi follow the steps to reproduce correctly please.

**Comment by danielnakamashi** [comment]
Yes, you're right. It's not working. I must have done something wrong when reproducing it.

**Comment by shafkathullah** [comment]
@danielnakamashi i think this behaviour is intended. otherwise for big form it would trigger whole form validation, which would be undesirable. i think i used `trigger` function to overcome at that time.

**Comment by shafkathullah** [comment]
or we can have a new api, like `validationDeps` array in `<Controller />` or `useForm` props? which would only validate sub set of depended fields?
